### ISSUE TITLE
Active response script fails for Windows endpoint

### ISSUE THREAD

**Issue by fdalmaup** [wazuh/wazuh#21812]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.8.0-beta1 | Active Response | Agent | Packages | Windows 10 |

During https://github.com/wazuh/wazuh/issues/21765, it was found that the active response `netsh` script for Windows endpoints is not blocking incoming requests for registered malicious IPs. However, the manager is registering the expected alerts [shown in our documentation](https://documentation-dev.wazuh.com/v4.8.0-beta1/proof-of-concept-guide/block-malicious-actor-ip-reputation.html#visualize-the-alerts). 
<details><summary>alerts.log</summary>

```log
Rule: 657 (level 3) -> 'Active response: active-response/bin/netsh.exe - add'
2024/02/08 11:05:32 active-response/bin/netsh.exe: {"version":1,"origin":{"name":"node01","module":"wazuh-execd"},"command":"add","parameters":{"extra_args":[],"alert":{"timestamp":"2024-02-08T19:05:30.347+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":7,"mail":false,"groups":["attack"]},"agent":{"id":"004","name":"DESKTOP-LURLNN8","ip":"172.31.15.155"},"manager":{"name":"ip-172-31-95-205"},"id":"1707419130.5458207","full_log":"172.31.33.48 - - [08/Feb/2024:11:05:30 -0800] \"GET / HTTP/1.1\" 200 46","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"172.31.33.48","id":"200","url":"/"},"location":"C:\\Apache24\\logs\\access.log"},"program":"active-response/bin/netsh.exe"}}
parameters.program: active-response/bin/netsh.exe
Rule: 657 (level 3) -> 'Active response: active-response/bin/netsh.exe - delete'
2024/02/08 11:06:33 active-response/bin/netsh.exe: {"version":1,"origin":{"name":"node01","module":"wazuh-execd"},"command":"delete","parameters":{"extra_args":[],"alert":{"timestamp":"2024-02-08T19:04:48.188+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":4,"mail":false,"groups":["attack"]},"agent":{"id":"004","name":"DESKTOP-LURLNN8","ip":"172.31.15.155"},"manager":{"name":"ip-172-31-95-205"},"id":"1707419088.5426111","full_log":"172.31.33.48 - - [08/Feb/2024:11:04:46 -0800] \"GET / HTTP/1.1\" 200 46","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"172.31.33.48","id":"200","url":"/"},"location":"C:\\Apache24\\logs\\access.log"},"program":"active-response/bin/netsh.exe"}}
```

</details>

It seems that the command to execute the script is sent but it does not block incoming connections from the same endpoint.

On the agent's side, no logs were found in `ossec-agent\logs` and on the manager's side, there were no logs in `/var/ossec/logs/active-responses.log`.

Every Agent and the manager were deployed in EC2 instances having defined Firewall rules.

More information in [this issue comment](https://github.com/wazuh/wazuh/issues/21765#issuecomment-1932528790).

**Comment by LucioDonda** [comment]
After using the same environment as in the test we found that the Active Response script wasn't failing, what actually happened was that the firewall wasn't active. In the active response logs we also notice some messages showing this.

### Failing Scenario:

Logs from the RHEL host without Firewall:
``` bash
[root@ip-172-31-33-48 ec2-user]# echo $(date +"%T") && curl http://172.31.15.155 
15:58:00
<html><body><h1>It works!</h1></body></html>
[root@ip-172-31-33-48 ec2-user]# echo $(date +"%T") && curl http://172.31.15.155 
15:58:03
<html><body><h1>It works!</h1></body></html>
[root@ip-172-31-33-48 ec2-user]# echo $(date +"%T") && curl http://172.31.15.155 
15:58:04
<html><body><h1>It works!</h1></body></html>
[root@ip-172-31-33-48 ec2-user]# echo $(date +"%T") && curl http://172.31.15.155 
15:58:05
<html><body><h1>It works!</h1></body></html>
[root@ip-172-31-33-48 ec2-user]# echo $(date +"%T") && curl http://172.31.15.155 
15:58:07
<html><body><h1>It works!</h1></body></html>
[root@ip-172-31-33-48 ec2-user]# echo $(date +"%T") && curl http://172.31.15.155 
15:58:08
<html><body><h1>It works!</h1></body></html>

```
WIndows Host:
![image](https://github.com/wazuh/wazuh/assets/95248059/69ec1cb8-0679-4f95-8c96-9f1bdbef5e8f)

Windows agent active response logs, (Although the rule has been added, it has no impact because the firewall is not active):
``` log
2024/02/09 07:58:02 active-response/bin/netsh.exe: Starting
2024/02/09 07:58:02 active-response/bin/netsh.exe: {"version":1,"origin":{"name":"node01","module":"wazuh-execd"},"command":"add","parameters":{"extra_args":[],"alert":{"timestamp":"2024-02-09T15:58:02.120+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":15,"mail":false,"groups":["attack"]},"agent":{"id":"004","name":"DESKTOP-LURLNN8","ip":"172.31.15.155"},"manager":{"name":"ip-172-31-95-205"},"id":"1707494282.7842543","full_log":"172.31.33.48 - - [09/Feb/2024:07:58:00 -0800] \"GET / HTTP/1.1\" 200 46","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"172.31.33.48","id":"200","url":"/"},"location":"C:\\Apache24\\logs\\access.log"},"program":"active-response/bin/netsh.exe"}}

2024/02/09 07:58:02 active-response/bin/netsh.exe: {"version":1,"origin":{"name":"netsh.exe","module":"active-response"},"command":"check_keys","parameters":{"keys":["172.31.33.48"]}}
2024/02/09 07:58:02 active-response/bin/netsh.exe: {"version":1,"origin":{"name":"node01","module":"wazuh-execd"},"command":"continue","parameters":{"extra_args":[],"alert":{"timestamp":"2024-02-09T15:58:02.120+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":15,"mail":false,"groups":["attack"]},"agent":{"id":"004","name":"DESKTOP-LURLNN8","ip":"172.31.15.155"},"manager":{"name":"ip-172-31-95-205"},"id":"1707494282.7842543","full_log":"172.31.33.48 - - [09/Feb/2024:07:58:00 -0800] \"GET / HTTP/1.1\" 200 46","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"172.31.33.48","id":"200","url":"/"},"location":"C:\\Apache24\\logs\\access.log"},"program":"active-response/bin/netsh.exe"}}

2024/02/09 07:58:02 active-response/bin/netsh.exe: {"message":"Active response may not have an effect","firewall":{"profile1":"FIREWALL_DOMAIN","status1":"active","profile2":"FIREWALL_PRIVATE","status2":"inactive","profile3":"FIREWALL_PUBLIC","status3":"inactive" },"status":"inactive","script":"netsh"}
2024/02/09 07:58:02 active-response/bin/netsh.exe: Ended
```


### Success Scenario:

Logs from the RHEL host with Firewall active (after more than a minute the firewall block goes away):
``` bash
[root@ip-172-31-33-48 ec2-user]# echo $(date +"%T") && curl http://172.31.15.155 
15:48:24
<html><body><h1>It works!</h1></body></html>
[root@ip-172-31-33-48 ec2-user]# echo $(date +"%T") && curl http://172.31.15.155 
15:48:26
^C
[root@ip-172-31-33-48 ec2-user]# echo $(date +"%T") && curl http://172.31.15.155 
15:48:29
^C
[root@ip-172-31-33-48 ec2-user]# echo $(date +"%T") && curl http://172.31.15.155 
15:49:27
^C
[root@ip-172-31-33-48 ec2-user]# echo $(date +"%T") && curl http://172.31.15.155 
15:49:29
<html><body><h1>It works!</h1></body></html>

```

Windows Host, firewall active and wazuh rule with propper IP:
![Captura desde 2024-02-09 13-00-01 (1)](https://github.com/wazuh/wazuh/assets/95248059/7fff8a3b-7804-4b72-b373-aece457ce4b8)

Active response logs on the windows agent (Adding and deleting the rule after the 60 seconds):

``` log
2024/02/09 07:48:26 active-response/bin/netsh.exe: Starting
2024/02/09 07:48:26 active-response/bin/netsh.exe: {"version":1,"origin":{"name":"node01","module":"wazuh-execd"},"command":"add","parameters":{"extra_args":[],"alert":{"timestamp":"2024-02-09T15:48:25.542+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":12,"mail":false,"groups":["attack"]},"agent":{"id":"004","name":"DESKTOP-LURLNN8","ip":"172.31.15.155"},"manager":{"name":"ip-172-31-95-205"},"id":"1707493705.7624507","full_log":"172.31.33.48 - - [09/Feb/2024:07:48:24 -0800] \"GET / HTTP/1.1\" 200 46","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"172.31.33.48","id":"200","url":"/"},"location":"C:\\Apache24\\logs\\access.log"},"program":"active-response/bin/netsh.exe"}}

2024/02/09 07:48:26 active-response/bin/netsh.exe: {"version":1,"origin":{"name":"netsh.exe","module":"active-response"},"command":"check_keys","parameters":{"keys":["172.31.33.48"]}}
2024/02/09 07:48:26 active-response/bin/netsh.exe: {"version":1,"origin":{"name":"node01","module":"wazuh-execd"},"command":"continue","parameters":{"extra_args":[],"alert":{"timestamp":"2024-02-09T15:48:25.542+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":12,"mail":false,"groups":["attack"]},"agent":{"id":"004","name":"DESKTOP-LURLNN8","ip":"172.31.15.155"},"manager":{"name":"ip-172-31-95-205"},"id":"1707493705.7624507","full_log":"172.31.33.48 - - [09/Feb/2024:07:48:24 -0800] \"GET / HTTP/1.1\" 200 46","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"172.31.33.48","id":"200","url":"/"},"location":"C:\\Apache24\\logs\\access.log"},"program":"active-response/bin/netsh.exe"}}

2024/02/09 07:48:26 active-response/bin/netsh.exe: Ended
```

``` log
2024/02/09 07:49:28 active-response/bin/netsh.exe: Starting
2024/02/09 07:49:28 active-response/bin/netsh.exe: {"version":1,"origin":{"name":"node01","module":"wazuh-execd"},"command":"delete","parameters":{"extra_args":[],"alert":{"timestamp":"2024-02-09T15:48:25.542+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":12,"mail":false,"groups":["attack"]},"agent":{"id":"004","name":"DESKTOP-LURLNN8","ip":"172.31.15.155"},"manager":{"name":"ip-172-31-95-205"},"id":"1707493705.7624507","full_log":"172.31.33.48 - - [09/Feb/2024:07:48:24 -0800] \"GET / HTTP/1.1\" 200 46","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"172.31.33.48","id":"200","url":"/"},"location":"C:\\Apache24\\logs\\access.log"},"program":"active-response/bin/netsh.exe"}}

2024/02/09 07:49:28 active-response/bin/netsh.exe: Ended
```

c.c. : @cborla @fdalmaup

**Comment by cborla** [comment]
Above test demonstrated that the firewall had to be enabled for the AR to run correctly. In addition, there are no changes between versions and the [previous test](https://github.com/wazuh/wazuh/issues/21360#issuecomment-1889817863) passed correctly. 

By analyzing the documentation, section [Windows endpoint](https://documentation-dev.wazuh.com/v4.8.0-beta1/proof-of-concept-guide/block-malicious-actor-ip-reputation.html#install-the-apache-web-server
), it does not seem to be necessary to update it, we can indirectly interpret that the firewall has to be enabled.

```
Navigate to the C:\Apache24\bin\ folder and run the following command in a PowerShell terminal with administrator privileges:

> .\httpd.exe
The first time you run the Apache binary a Windows Defender Firewall pops up.

Click on Allow Access. This allows the Apache HTTP server to communicate on your private or public networks depending on your network setting. It creates an inbound rule in your firewall to allow incoming traffic on port 80.
```


LGTM!

**Comment by sito0984** [comment]
Hi, I believe I found a reproducible Windows Active Response problem on Wazuh Agent 4.14.7 that may be related to this issue.

In my case, the Active Response request reaches the Windows agent correctly and contains a valid parameters.alert.data.srcip, but the stock netsh.exe helper logs:

Cannot read 'srcip' from data

and no Windows Firewall block rule is created.

I traced the failure in the official Wazuh v4.14.7 source to the IP validation path:

get_srcip_from_json() → get_ip_version() → getaddrinfo(..., AI_NUMERICHOST, ...)

On Windows, this path calls getaddrinfo() without initializing Winsock first.

In a direct Windows test, valid IPv4 and IPv6 addresses returned WSANOTINITIALISED (10093) before WSAStartup(). After calling WSAStartup(MAKEWORD(2,2)) before the validation and WSACleanup() afterward, valid IPv4/IPv6 addresses were accepted correctly while invalid addresses were still rejected.

I then built a locally patched Wazuh 4.14.7 netsh.exe and tested it end-to-end on Windows. The full Manager → Agent → Active Response helper → Windows Firewall flow worked successfully: the IP was blocked, and the rule was automatically removed after the configured 600-second timeout.

The patch only adds Winsock initialization/cleanup around the Windows getaddrinfo() validation path; the non-Windows path is unchanged.

I can provide the minimal patch and reproduction/test details if they would be useful.

Could you confirm whether this behavior is already fixed in an upcoming Wazuh release, or whether this should be tracked as a current Windows Active Response bug?
